.github/workflows/src/summarize-checks.js

```diff
@@ -0,0 +1,856 @@
+  {
+    precedence: 6,
+    name: "SDK azure-sdk-for-java",
+    suppressionLabels: [],
+    troubleshootingGuide: checkAndDiagramTsg(3),
+  },
+  {
+    precedence: 6,
+    name: "SDK azure-sdk-for-js",
+    suppressionLabels: [],
+    troubleshootingGuide: checkAndDiagramTsg(3),
+  },
+  {
+    precedence: 6,
+    name: "SDK azure-sdk-for-python",
+    suppressionLabels: [],
+    troubleshootingGuide: checkAndDiagramTsg(3),
+  },
+  {
+    precedence: 6,
+    name: "SDK azure-sdk-for-python-track2",
+    suppressionLabels: [],
+    troubleshootingGuide: checkAndDiagramTsg(3),
+  },
+  {
+    precedence: 10,
+    name: AUTOMATED_CHECK_NAME,
+    suppressionLabels: [],
+    troubleshootingGuide: reqMetCheckTsg,
+  },
+];
+
+// during renderAutomatedMergingRequirementsMetCheck we resolve the result of
+// automated merge requirements met by from the result of and(requiredChecks).
+// if any are pending, automated merging requirements is pending. This is ripe for complete removal
+// in favor of just honoring the `required` checks results directly.
+/** @type {string[]} */
+const EXCLUDED_CHECK_NAMES = [];
+
+// #endregion
+// #region core
+/**
+ * @param {import('github-script').AsyncFunctionArguments} AsyncFunctionArguments
+ * @returns {Promise<void>}
+ */
+export default async function summarizeChecks({ github, context, core }) {
+  logGitHubRateLimitInfo({ github, context, core });
+  let { owner, repo, issue_number, head_sha } = await extractInputs(github, context, core);
+  const targetBranch = context.payload.pull_request?.base?.ref;
+  core.info(`PR target branch: ${targetBranch}`);
+
+  await summarizeChecksImpl(
+    {
+      github,
+      context,
+      core,
+    },
+    owner,
+    repo,
+    issue_number,
+    head_sha,
+    context.event_name,
+    targetBranch,
+  );
+}
+
+/**
+ * @param {import('github-script').AsyncFunctionArguments} AsyncFunctionArguments
+ * @param {string} owner
+ * @param {string} repo
+ * @param {number} issue_number
+ * @param {string} head_sha
+ * @param {string} event_name
+ * @param {string} targetBranch
+ * @returns {Promise<void>}
+ */
+export async function summarizeChecksImpl(
+  { github, context, core },
+  owner,
+  repo,
+  issue_number,
+  head_sha,
+  event_name,
+  targetBranch,
+) {
+  /** @type {string[]} */
+  let labelNames = [];
+  /** @type {CheckRunData[]} */
+  let requiredCheckRuns = [];
+  /** @type {CheckRunData[]} */
+  let fyiCheckRuns = [];
+
+  core.info(
+    `Handling ${event_name} event for PR #${issue_number} in ${owner}/${repo} with targeted branch ${targetBranch}`,
+  );
+
+  // no matter what, we will need the labels for the PR, so let's fetch them first
+  try {
+    const resp = await github.rest.issues.listLabelsOnIssue({
+      owner,
+      repo,
+      issue_number,
+    });
+    /** @type {{ name: string }[]} */
+    const labels = resp.data;
+    /** @type {string[]} */
+    labelNames = labels.map((label) => label.name);
+  } catch (error) {
+    core.error(
```

Comment:
What if we don't catch, and just let the exception bubble up?  Is this good enough?

If we do catch and throw a custom Error, we should try to wrap the existing error.

Reply:
Incoming commit.